Slightly reword installation guide to attempt to stop people using dev

Vexed01 opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Make using the dev version more heavily worded in an attempt to stop people from using it as an normal end user.

Vexed01 committed:
create

Vexed01 committed:
add the code

Vexed01 committed:
change de logs

Vexed01 committed:
lets remember linux exists

Vexed01 committed:
fix formtting and change wording

Vexed01 committed:
change the wording a bit more... after running `sphinx-build` it def looks like i used bold a lot

god that was a long commit mesage

Flame442 commented on docs/install_linux_mac.rst:
```diff
@@ -286,8 +286,10 @@ Or, to install with PostgreSQL support:
-
+  link below. Please note that **dev is experimental and not intended for end users.** We will not
+  support anyone using the development version in any support channels. Downgrading after an
+  accidental install of dev is not possible without data loss.
```
It might be better to say "not __always__ possible".

Drapersniper replied:
Only replying cuz I’m nosey, I do agree but also disagree if we don’t want end users to use dev since it usually has breaking changes/incomplete changes etc and often leaves the user unable to downgrade, we shouldn’t imply that sometimes they will get support although it will happen we probably shouldn’t imply here, if we take an official stance that we will not help it should discourage people

mikeshardmind replied:
I'd prefer the more accurate wording here. It doesn't change that we don't provide support for development versions, only make it clear what risks might be associated with using that version.

Flame442 commented on docs/install_linux_mac.rst:
```diff
@@ -286,8 +286,10 @@ Or, to install with PostgreSQL support:
   To install the development version, replace ``Red-DiscordBot`` in the above commands with the
-  following link:
-
+  link below. Please note that **dev is experimental and not intended for end users.** We will not
```
This sentence should be reworded to something along the lines of "The development version of the bot contains experimental changes. It is not intended for normal use." (all in bold)

Flame442 commented on docs/install_windows.rst:
```diff
@@ -91,7 +91,9 @@ Installing Red
-      following link:
+      link below. Please note that **dev is experimental and not intended for end users.** We will not
+      support anyone using the development version in any support channels. Downgrading after an
+      accidental install of dev is not possible without data loss.
```
Changes from above apply.

Vexed01 committed:
review changes

Vexed01 commented:
I'm not happy with how long this is getting when built:


![image](https://user-images.githubusercontent.com/51716387/69477249-c9a11f00-0ddb-11ea-8b27-3c00add07777.png)


Should we just leave it as:
>To install the development version, replace ``Red-DiscordBot`` in the above commands with the
link below. **The development version of the bot contains experimental changes. It is not
intended for normal users.** We will not support you with any problems that may arise with the core bot or third party cogs, and downgrading to stable may cause data loss.

I think the above gets out the message in as few words as possible.

Drapersniper commented on docs/install_windows.rst:
```diff
@@ -91,9 +91,11 @@ Installing Red
+      intended for normal users.** We will not support anyone using the development version in any
+      support channels. Using the develepment version may break third party cogs and not all core
+      commands may work. Downgrading after an accidental install of dev is not always possible
+      without data loss.
```
Data loss is not the only issue here it could also cause unexpected crashes

Drapersniper commented:
> I'm not happy with how long this is getting when built:
> 
> ![image](https://user-images.githubusercontent.com/51716387/69477249-c9a11f00-0ddb-11ea-8b27-3c00add07777.png)
> 
> Should we just leave it as:
> 
> > To install the development version, replace `Red-DiscordBot` in the above commands with the
> > link below. **The development version of the bot contains experimental changes. It is not
> > intended for normal users.** We will not support you with any problems that may arise with the core bot or third party cogs, and downgrading to stable may cause data loss.
> 
> I think the above gets out the message in as few words as possible.

I would say since it’s the docs it’s better to be explicit rather than try to make the message as short as possible, there no reason to try to shorter the docs after all their purpose is to educate the end user if the end user doesn’t read it because it’s too long it’s their fault

Vexed01 committed:
draper's review (see description)

i have put or worse partially to scare users and partially as there are other possible effects of a downgrade

Flame442 commented on docs/install_linux_mac.rst:
```diff
@@ -286,7 +286,11 @@ Or, to install with PostgreSQL support:
-  following link:
+  link below. **The development version of the bot contains experimental changes. It is not
+  intended for normal users.** We will not support anyone using the development version in any
+  support channels. Using the develepment version may break third party cogs and not all core
```
Typo, "develepment".

Flame442 commented on docs/install_linux_mac.rst:
```diff
@@ -286,7 +286,11 @@ Or, to install with PostgreSQL support:
+  link below. **The development version of the bot contains experimental changes. It is not
+  intended for normal users.** We will not support anyone using the development version in any
+  support channels. Using the develepment version may break third party cogs and not all core
+  commands may work. Downgrading after an accidental install of dev may cause data loss, crashes
```
I don't think we should say "after an accidental install of dev", since even purposeful installs have these side effects. Maybe "Downgrading after installing the development version may..."

Flame442 commented on docs/install_windows.rst:
```diff
@@ -91,7 +91,11 @@ Installing Red
 
       To install the development version, replace ``Red-DiscordBot`` in the above commands with the
-      following link:
+      link below. **The development version of the bot contains experimental changes. It is not
```
Same as above.

Vexed01 committed:
review

Vexed01 committed:
clarity
